### PR TITLE
Support correlation ID in pre-execution

### PR DESCRIPTION
**Pull request by yontyon**
This PR introduces correlation ID support in pre-execution.
* Adding correlation ID to PreProcessReplyMsg.
* Adding correlation ID logging when receiving messages and in the background threads
* Fix a little bug in pre-execution

### CODE REVIEW

File: bftengine/src/preprocessor/messages/PreProcessReplyMsg.cpp [hunk 1]
```diff
@@ -66,8 +66,13 @@ void PreProcessReplyMsg::setupMsgBody(const char* buf, uint32_t bufLen) {
 
   // Sign hash
   sigManager_->sign((char*)hash.data(), SHA3_256::SIZE_IN_BYTES, body() + headerSize, sigSize);
-  msgSize_ = headerSize + sigSize;
+  memcpy(body() + headerSize + sigSize, cid.c_str(), cid.size());
+  msgBody()->cidLength = cid.size();
+  msgSize_ = headerSize + sigSize + msgBody()->cidLength;
   msgBody()->replyLength = sigSize;
```
Comment (yuliasherman) [inline bftengine/src/preprocessor/messages/PreProcessReplyMsg.cpp]: Hm. Looks like we don't take into account cid. Let's talk about it.

Comment (yuliasherman) [reply bftengine/src/preprocessor/messages/PreProcessReplyMsg.cpp]: During a discussion, it was clarified that replyLength contains only signed pre-executed hash and does not contain cid length.

File: bftengine/src/preprocessor/PreProcessor.cpp [hunk 2]
```diff
@@ -469,13 +471,13 @@ void PreProcessor::handleReqPreProcessingJob(PreProcessRequestMsgSharedPtr prePr
   uint32_t actualResultBufLen =
       launchReqPreProcessing(clientId, reqSeqNum, preProcessReqMsg->requestLength(), preProcessReqMsg->requestBuf());
   if (isPrimary && isRetry) {
-    handlePreProcessedReqPrimaryRetry(clientId, reqSeqNum);
+    handlePreProcessedReqPrimaryRetry(clientId, reqSeqNum, preProcessReqMsg->getCid());
```
[the remark this answers is not in the document]
Comment (yontyon) [reply bftengine/src/preprocessor/PreProcessor.cpp]: I will have to use a mutex to access the clientEntry and I prefer to use as fewer locks as I can

[the remark this answers is not in the document]
Comment (yuliasherman) [reply bftengine/src/preprocessor/PreProcessor.cpp]: Both functions called from handlePreProcessedReqPrimaryRetry already take the lock, so you will not add additional overhead.

File: bftengine/src/preprocessor/PreProcessor.cpp [hunk 3]
```diff
@@ -321,7 +320,7 @@ void PreProcessor::finalizePreProcessing(NodeIdType clientId, const std::string
                                                      reqSeqNum,
                                                      clientEntry->clientReqInfoPtr->getPrimaryPreProcessedResultLen(),
                                                      clientEntry->clientReqInfoPtr->getPrimaryPreProcessedResult(),
-                                                     clientRequestMsg->requestTimeoutMilli(),
+                                                     1000,
```
[the remark this answers is not in the document]
Comment (yontyon) [reply bftengine/src/preprocessor/PreProcessor.cpp]: Fixed

File: bftengine/src/preprocessor/RequestProcessingInfo.hpp [hunk 4]
```diff
@@ -42,7 +42,7 @@ class RequestProcessingInfo {
   const char* getPrimaryPreProcessedResult() const { return primaryPreProcessResult_; }
   uint32_t getPrimaryPreProcessedResultLen() const { return primaryPreProcessResultLen_; }
   bool isReqTimedOut() const;
-
+  uint64_t getReqTimeoutMilli() { return clientPreProcessReqMsg_->requestTimeoutMilli(); }
```
Comment (yuliasherman) [inline bftengine/src/preprocessor/RequestProcessingInfo.hpp]: Make it const function

File: bftengine/src/preprocessor/PreProcessor.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (yontyon) [reply bftengine/src/preprocessor/PreProcessor.cpp]: Done

File: bftengine/src/preprocessor/RequestProcessingInfo.hpp (hunk 4 above)
Comment (yontyon) [reply bftengine/src/preprocessor/RequestProcessingInfo.hpp]: Done

**Pull request merged**